int32 id
string behaviour
int32 topology_id
int32 shape_id